lib/ecto/model/timestamps.ex

```diff
@@ -19,31 +19,42 @@ defmodule Ecto.Model.Timestamps do
   @doc """
   Puts a timestamp in the changeset with the given field and type.
   """
-  def put_timestamp(changeset, field, type) do
+  def put_timestamp(changeset, field, type, use_usec) do
     if get_change changeset, field do
       changeset
     else
-      {date, {h, m, s}} = :erlang.universaltime
-      put_change changeset, field, Ecto.Type.load!(type, {date, {h, m, s, 0}})
+      put_change changeset, field, Ecto.Type.load!(type, timestamp_tuple(use_usec))
     end
   end
 
+  defp timestamp_tuple(use_usec) when use_usec == true do
```

Comment:
You can match directly: `timestamp_tuple(true)` or `timestamp_tuple(_use_usec = true)`.